staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -144,7 +157,7 @@ func (s *SpdyRoundTripper) dial(req *http.Request) (net.Conn, error) {
 		proxyReq.Header.Set("Proxy-Authorization", pa)
 	}
 
-	proxyDialConn, err := s.dialWithoutProxy(req.Context(), proxyURL)
+	proxyDialConn, err := s.dialWithoutProxy(proxyReq.Context(), proxyURL)
```

Comment:
`proxyReq.Context()` is detached from the original req.Context(), which means we lose timeout/cancellation threading

Reply:
Passed req into dialWithHttpProxy and used req.WithContext to assume the original request context.